gazebo_state_plugins/src/GazeboObjectInfo.cpp

```diff
@@ -211,23 +313,31 @@ GazeboObjectInfo::ObjectMsg GazeboObjectInfo::createBoundingBoxObject(physics::M
             pose.orientation.z=coll_pose.rot.z;
             pose.orientation.w=coll_pose.rot.w;
 
-            obj.primitive_poses.push_back(pose);
-        
+            if(c->GetShape()->HasType(gazebo::physics::Base::MESH_SHAPE)) obj.mesh_poses.push_back(pose);
+            if(c->GetShape()->HasType(gazebo::physics::Base::BOX_SHAPE)) obj.primitive_poses.push_back(pose);
+
             if (!origin_init)
             {
                 obj.origin = pose;
             }
-    
-            if (include_shape) {
-
-                shape_msgs::SolidPrimitive * solid=getSolidPrimitive(c);
-                if (!solid) {
-                    ROS_WARN("Skipping coll %s of link %s of model %s, could not get SolidPrimitive. ",c->GetName().c_str(),linkName.c_str(),obj.name.c_str());
-                    continue;
-                }
 
-                obj.primitives.push_back(*solid);
-                delete solid;
+            if (include_shape) {
+            	if(c->GetShape()->HasType(gazebo::physics::Base::MESH_SHAPE)){
```

Comment:
can you fix formatting here, no tabs and keep the amount of spaces the same throughout the document